### DISCUSSION TITLE
Custom brushes not working as expected on mobile 4.6.0

### DISCUSSION THREAD

**Discussion by kawka-maciej** [fabricjs/fabric.js#7815] [General]
Hello! In my project I was trying to use some custom Fabric brushes from [here](https://github.com/tennisonchan/fabric-brush/blob/master/src/brushes/marker_brush.js). They work on the desktop but the problem appears to be on the mobile. Below are some gifs to demonstrate the difference
Default pencil brush:
![normal](https://user-images.githubusercontent.com/59606418/158634837-f0da7c3e-019e-437b-a4ff-b1985582bb60.gif)
Custom brush:
![custom](https://user-images.githubusercontent.com/59606418/158634851-25b9e8b0-6ac1-4624-b705-f97c8f13e3d5.gif)
As you can see, the final drawings is moved. Could anyone hint me what needs to be adjusted in order for the custom brush to work on mobile as well?

**Comment by kawka-maciej** [comment]
I browsed some code of the default brushes in Fabric.js and came to the conclusion that to ensure the same behaviour across desktop and mobile, the custom brush that I mentioned above needs to be adjusted to use the Fabric API directly to draw the brush path. 

The current problem with the implementation there was that it used the canvas context directly to draw the brush path, which caused the weird behaviour on mobile (as on the gif above). The default implementations that I browsed uses a different strategy: the preview path was drawn directly to the canvas (one of the two that Fabric uses) and then the canvas is cleared and the changes are applied directly using Fabric API ( as [here](https://github.com/fabricjs/fabric.js/blob/master/src/brushes/spray_brush.class.js#L96-L118), the spray path is recreated using fabric.Rect objects, the used context is cleared.). I adjusted my implementation to use the strategy and it seems to work. Although I still don't know where this difference in behaviour comes from.

**Comment by ShaMan123** [comment]
look into retina scaling
